bench/src/bulk.rs

```diff
@@ -80,34 +128,112 @@ async fn client(server_addr: SocketAddr, server_cert: quinn::Certificate) -> Res
     client_config
         .add_certificate_authority(server_cert)
         .unwrap();
+    let mut client_config = client_config.build();
+    client_config.transport = Arc::new(transport_config());
+
     let quinn::NewConnection { connection, .. } = endpoint
-        .connect_with(client_config.build(), &server_addr, "localhost")
+        .connect_with(client_config, &server_addr, "localhost")
         .unwrap()
         .await
         .context("unable to connect")?;
     trace!("connected");
 
-    let mut stream = connection
-        .open_uni()
-        .await
-        .context("failed to open stream")?;
-    const DATA: &[u8] = &[0xAB; 1024 * 1024];
     let start = Instant::now();
-    for _ in 0..1024 {
-        stream
-            .write_all(DATA)
-            .await
-            .context("failed sending data")?;
+
+    let connection = Arc::new(connection);
+    let get_connection = || connection.clone();
+
+    let mut ops =
+        futures::stream::iter((0..opt.streams).map(|_| async move {
+            send_data_on_stream(get_connection(), opt.stream_size_mb).await
+        }))
```

Comment:
Something like this?

```suggestion
        futures::stream::iter((0..opt.streams).map(|_| {
            let connection = connection.clone();
            async move {
                send_data_on_stream(connection, opt.stream_size_mb).await
            }
        }))
```

Reply:
I don't think that is any improvement (as a reader now the first thing I'm asking myself is what this weird extra block is doing there), but I still applied it.

Frankly speaking: I feel there is a bit of too much bikeshedding going on. This comments all lead to spending effort to move code back and forth without any objective reason why it should be in one particular place, and without any functional changes. If this is highly important to you, please consider writing a styleguide

Reply:
That's fair, I'll see if I can write a style guide soonish. On the other hand, I'm not sure it would have covered everything we commented on here. In my experience, we routinely we give each other feedback on the clearest way to express something in code/submit something for review, and make changes accordingly, and it generally has been fine.